drivers/hmis/app/models/hmis/project_auto_exit_config.rb

```diff
@@ -5,6 +5,37 @@
 ###
 
 class Hmis::ProjectAutoExitConfig < Hmis::ProjectConfig
-  # This class is not yet used, but will be replacing AutoExitConfig
-  # TODO: Validate that config_options is not null and length_of_absence_days is defined on that json blob
+  validate :length_of_absence_days_ge_30
+
+  def length_of_absence_days=(value)
+    self.config_options = { 'length_of_absence_days': value }.to_json
+  end
+
+  def length_of_absence_days
+    JSON.parse(config_options)['length_of_absence_days']
+  end
+
+  def length_of_absence_days_ge_30
+    unless config_options
+      errors.add(:base, 'config_options must be present')
+      return
+    end
+
+    begin
+      json_blob = JSON.parse(config_options)
+    rescue JSON::ParserError
+      errors.add(:base, 'config_options must be JSON')
```

Comment:
I think this can be validated on the parent model, since we know it should always be JSON.

I like the approach todd took here, with some helpers for reading and setting the json field.
https://github.com/greenriver/hmis-warehouse/blob/release-105/app/models/grda_warehouse/remote_credentials/oauth.rb

So maybe on the Hmis::ProjectConfig model:
```suggestion
  validate :validate_config_options_json

  def validate_config_options_json
    return unless config_options

    begin
      JSON.parse(config_options)
    rescue JSON::ParserError
      errors.add(:base, 'config_options must be JSON')
    end
  end

  def options= hash
    self.config_options = hash.to_json
  end

  def options
    @options ||= JSON.parse(config_options)
  rescue JSON::ParserError, TypeError
    nil
  end
```

Then this validator here can simply read the value from `options` hash and validate it as you do:
```
length_of_absence_days = options['length_of_absence_days']
```

Reply:
Okay sounds good!

This line was giving me trouble:
`@options ||= JSON.parse(config_options)`
Creating an `options` instance variable I think was causing `options` and `config_options` to get out of sync, so for instance (in the rails console),
```
irb(main):024:0> aec.options = {"length_of_absence_days": 40}
=> {:length_of_absence_days=>40}
irb(main):025:0> aec.options
=> {"length_of_absence_days"=>40}
irb(main):026:0> aec.config_options
=> "{\"length_of_absence_days\":40}"
irb(main):027:0> aec.config_options = '{"foo":"bar"}'
=> "{\"foo\":\"bar\"}"
irb(main):030:0> aec.config_options
=> "{\"foo\":\"bar\"}"
irb(main):029:0> aec.options
=> {"length_of_absence_days"=>40}
```

I've instead just made options return `JSON.parse(config_options)` so there is only 1 source of truth and no other info being stored. Let me know if this looks ok.

Reply:
Sounds great, thanks